ngraph/core/CMakeLists.txt

```diff
@@ -119,14 +119,12 @@ if (NOT WIN32)
     target_link_libraries(ngraph PRIVATE dl)
 endif()
 
-# Build subdirectories for all build types on Windows
-if(WIN32)
-    foreach(BUILD_TYPE Release Debug RelWithDebInfo MinSizeRel)
-        if(NOT EXISTS ${NGRAPH_BUILD_DIR}/${BUILD_TYPE})
-            file(MAKE_DIRECTORY ${NGRAPH_BUILD_DIR}/${BUILD_TYPE})
-        endif()
-    endforeach()
-endif()
+# Create sub-directories for all build types in case of multi-configuration generators
+foreach(BUILD_TYPE IN LISTS CMAKE_CONFIGURATION_TYPES)
```

Comment:
What if `CMAKE_CONFIGURATION_TYPES` is not defined?

Reply:
It will be dereferenced as empty list and the loop will be skipped.

Reply:
Can we just remove this cycle? I think it should affect nothing.

Reply:
Removed